subsys/bluetooth/mesh/pb_adv.c

```diff
@@ -317,6 +359,142 @@ static void protocol_timeout(struct k_work *work)
 	link.rx.seg = 0U;
 	prov_link_close(PROV_BEARER_LINK_STATUS_TIMEOUT);
 }
+
+static struct delayable_adv_ctx *delayable_adv_ctx_get_unused(void)
+{
+	for (int i = 0; i < CONFIG_BT_MESH_PB_ADV_DELAYABLE_ADV_COUNT; i++) {
+		if (!atomic_test_bit(delayable_advs_ctx[i].flags, DELAYABLE_ADV_CTX_IN_USE)) {
+			return &delayable_advs_ctx[i];
+		}
+	}
+
+	return NULL;
+}
+
+static struct delayable_adv_ctx *delayable_adv_ctx_assign(struct bt_mesh_adv *adv,
+							  const struct bt_mesh_send_cb *cb,
+							  void *cb_data, bool retransmit)
+{
+	struct delayable_adv_ctx *ctx;
+
+	ctx = delayable_adv_ctx_get_unused();
+	if (!ctx) {
+		LOG_WRN("Purge pending delayed advertisement.");
+		if (!delayable_adv_ctx_push()) {
+			return NULL;
+		}
+
+		ctx = delayable_adv_ctx_get_unused();
```

Comment:
Previous implementation was always able to send provisioning PDUs, no we can't. Can we do something with this?

Reply:
I do not understand what you mean by this?

Reply:
If purge busy context, it should be able always to allocate new one. It should be able always to send provisioning PDU.